src/net/socket_state.rs

```
@@ -129,7 +118,7 @@ impl SocketState {
 
     /// This will process an incoming packet and update acknowledgement information.
     pub fn process_received(&mut self, addr: SocketAddr, packet: &PacketHeader) -> Result<()>{
-        let connection = self.create_connection_if_not_exists(&addr)?;
+        let connection = self.connections.insert_if_not_exists(&addr)?;
         let mut lock = connection
             .write()
             .map_err(|_| NetworkError::AddConnectionToManagerFailed)?;
```

Should we bubble up this error?